src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -213,6 +200,21 @@ export class MissingTerms extends Component {
     }
   }
 
+  private isValidTerm(term) {
+    //p{L} is a Unicode script that matches any character in any language.
+    const wordWithBreakpoints = `\\p{L}*[-'?\*’.~=,\/\\\\:\`;_!&\(\)]+\\p{L}*`;
+    const regex = XRegExp(wordWithBreakpoints, 'gi');
+    const query = this.queryStateModel.get('q');
+    let words = regex.exec(query);
+    while (words !== null) {
+      if (words[0].indexOf(term) > -1) {
```

Comment:
It is not possible to use with exec. I will try with [match](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@match) 

We return false because we decided to always hide the term if is a part of a term broken by one of the special character from the regex